### PR TITLE
API doc for algorithm, message, value, time, and 2pc

### PR DESCRIPTION
**Pull request by vaporos**
Also removes <P> from Algorithm, which isn't a necessary generic.

### CODE REVIEW

File: libaugrim/src/algorithm/mod.rs [hunk 1]
```diff
@@ -21,11 +21,31 @@ use crate::error::{AlgorithmError, InternalError};
...
 pub trait Value: Clone {}
 
+/// A consensus algorithm.
+///
+/// An algorithm processes events with a given context, producing a set of actions.
+///
+/// Events are inputs into the algorithm. For example, if a message is received from another
+/// process, it is input into the algorithm as an event (likely a message delivery event).
+///
+/// Actions are the outputs of the algorithm. For example, the algorithm my output an action to
```
Comment (agunde406) [inline libaugrim/src/algorithm/mod.rs]: my -> may

File: libaugrim/src/lib.rs [hunk 2]
```diff
@@ -28,7 +28,6 @@
 //!
 //! The specific actions and events differ by algorithm; refer to the algorithm's documentation.
 
-
```
Comment (agunde406) [inline libaugrim/src/lib.rs]: This should be a part of the previous commit

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libaugrim/src/algorithm/mod.rs (hunk 1 above)
Comment (vaporos) [reply libaugrim/src/algorithm/mod.rs]: Fixed.

File: libaugrim/src/lib.rs (hunk 2 above)
Comment (vaporos) [reply libaugrim/src/lib.rs]: Fixed.

**Review by agunde406 — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**